### PR TITLE
fix: [CDS-54269]: refactor kustomize client to support cmd flags

### PR DESCRIPTION
**Pull request by achyuth-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)

### CODE REVIEW

File: 960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java [hunk 1]
```diff
@@ -42,29 +46,44 @@
...
 
   @Override
   public CliResponse buildWithPlugins(@Nonnull String manifestFilesDirectory, @Nonnull String kustomizeDirPath,
-      @Nonnull String kustomizeBinaryPath, @Nonnull String pluginPath, @Nonnull LogCallback executionLogCallback)
-      throws InterruptedException, TimeoutException, IOException {
+      @Nonnull String kustomizeBinaryPath, @Nonnull String pluginPath, @Nonnull LogCallback executionLogCallback,
+      Map<String, String> commandFlags) throws InterruptedException, TimeoutException, IOException {
+    String buildCommandFlag = "";
+    if (isNotEmpty(commandFlags)) {
+      buildCommandFlag = isEmpty(commandFlags.get(KustomizeCommands.buildCommand))
```
Comment (abosii-harness) [inline 960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java:74]: that can be just if commandFlags.containsKey() then buildCommandFlag = <value>

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Review by achyuth-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 8 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java` line 74 at r1](https://reviewable.io/reviews/harness/harness-core/44419#-NPI4DeZ3nTUFS7hi87Y-r1-74:-NPWjrgR5gsUD6UY5uMZ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1a364fc17d8c9990b51930f1bd8bf6e498b3399a/960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java#L74)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

that can be just if commandFlags.containsKey() then buildCommandFlag = <value>
</blockquote></details>

Done.

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 7 files at r1, 5 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 8 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/kustomize/KustomizeTaskHelper.java` line 63 at r2](https://reviewable.io/reviews/harness/harness-core/44419#-NPWjxmXBwAUGzKrkVhk:-NPWjxmYAe47Awtu8t3V:bhnpxqx) ([raw file](https://github.com/harness/harness-core/blob/fd3ba67651d423012bced8235c891753d2bca870/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/kustomize/KustomizeTaskHelper.java#L63)):*
> ```Java
>     // ToDo: set command-flags correctly
>     KustomizeClient kustomizeClient = kustomizeClientFactory.getClient(
>         manifestFilesDirectory, kustomizeBinaryPath, kustomizeDirPath, Collections.emptyMap());
> ```

Let's have only binary path and command flags as part of the getClient factory method
Manifest files directory and kustomizeDirPath is only specific to build commands, so theoretically there could be commands that doesn't require this
___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java` line 47 at r2](https://reviewable.io/reviews/harness/harness-core/44419#-NPWkXuA9NtbC-PO_30Z:-NPWkXuA9NtbC-PO_30_:bwevv11) ([raw file](https://github.com/harness/harness-core/blob/fd3ba67651d423012bced8235c891753d2bca870/960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java#L47)):*
> ```Java
>   private Version version;
>   private Map<String, String> commandFlags;
>   @Inject CliHelper cliHelper;
> ```

How this is injected? I do believe the factory method should inject it (which doesn't right now) so @Inject annotation is not required here
___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java` line 3 at r2](https://reviewable.io/reviews/harness/harness-core/44419#-NPWkrjG-MBaADhWa68E:-NPWkrjG-MBaADhWa68F:b-rnbkcx) ([raw file](https://github.com/harness/harness-core/blob/fd3ba67651d423012bced8235c891753d2bca870/960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java#L3)):*
> ```Java
> package io.harness.kustomize;
> 
> public final class KustomizeCommands {
> ```

Can this be a enum?

**Review by achyuth-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 9 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/kustomize/KustomizeTaskHelper.java` line 63 at r2](https://reviewable.io/reviews/harness/harness-core/44419#-NPWjxmXBwAUGzKrkVhk:-NPWuLKI-6HmKyRyxnlc:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/fd3ba67651d423012bced8235c891753d2bca870/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/kustomize/KustomizeTaskHelper.java#L63)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Let's have only binary path and command flags as part of the getClient factory method
Manifest files directory and kustomizeDirPath is only specific to build commands, so theoretically there could be commands that doesn't require this
</blockquote></details>

Done.
___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java` line 47 at r2](https://reviewable.io/reviews/harness/harness-core/44419#-NPWkXuA9NtbC-PO_30Z:-NPWuZqaEBF8rVqZfL8i:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/fd3ba67651d423012bced8235c891753d2bca870/960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java#L47)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

How this is injected? I do believe the factory method should inject it (which doesn't right now) so @Inject annotation is not required here
</blockquote></details>

Done.
___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java` line 3 at r2](https://reviewable.io/reviews/harness/harness-core/44419#-NPWkrjG-MBaADhWa68E:-NPWu_ZY7afPr7-V-Wdw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/fd3ba67651d423012bced8235c891753d2bca870/960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java#L3)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Can this be a enum?
</blockquote></details>

Done.

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 7 files at r2, 1 of 2 files at r3, 6 of 7 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 11 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeClientFactory.java` line 66 at r7](https://reviewable.io/reviews/harness/harness-core/44419#-NPq7JZq7_uKV7vBxtLD:-NPq7JZq7_uKV7vBxtLE:bm4utik) ([raw file](https://github.com/harness/harness-core/blob/a6583385efbc0600af0905779973ee7580f0d923/960-api-services/src/main/java/io/harness/kustomize/KustomizeClientFactory.java#L66)):*
> ```Java
>       }
> 
>     } catch (IOException | InterruptedException | TimeoutException e) {
> ```

InterruptedException should be catch in a different branch and Thread.currentThread().interrupt() should be used otherwise this may end in a orphan thread
___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java` line 43 at r7](https://reviewable.io/reviews/harness/harness-core/44419#-NPq7bJB0WNIT5QM8BtJ:-NPq7bJB0WNIT5QM8BtK:b-3v05fa) ([raw file](https://github.com/harness/harness-core/blob/a6583385efbc0600af0905779973ee7580f0d923/960-api-services/src/main/java/io/harness/kustomize/KustomizeClientImpl.java#L43)):*
> ```Java
>   private String kustomizeBinaryPath;
>   private Version version;
>   private Map<KustomizeCommands, String> commandFlags;
> ```

nit: I was thinking more about to keep use String but for reference use KustomizeCommands.BUILD.name(), but if you think it will work for you then I'm good, but you may have to transform from String to KustomizeCommand from request
___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java` line 10 at r7](https://reviewable.io/reviews/harness/harness-core/44419#-NPq7t1v46Pvud3nlAQ3:-NPq7t1v46Pvud3nlAQ4:b-toneow) ([raw file](https://github.com/harness/harness-core/blob/a6583385efbc0600af0905779973ee7580f0d923/960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java#L10)):*
> ```Java
> package io.harness.kustomize;
> 
> public enum KustomizeCommands {
> ```

Name should be KustomizeCommand

**Review by achyuth-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 11 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeClientFactory.java` line 66 at r7](https://reviewable.io/reviews/harness/harness-core/44419#-NPq7JZq7_uKV7vBxtLD:-NPqDxzp0A0B1L_zU18N:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a6583385efbc0600af0905779973ee7580f0d923/960-api-services/src/main/java/io/harness/kustomize/KustomizeClientFactory.java#L66)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

InterruptedException should be catch in a different branch and Thread.currentThread().interrupt() should be used otherwise this may end in a orphan thread
</blockquote></details>

Done.
___
*[`960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java` line 10 at r7](https://reviewable.io/reviews/harness/harness-core/44419#-NPq7t1v46Pvud3nlAQ3:-NPqF9n5-rC2APdhpKNZ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a6583385efbc0600af0905779973ee7580f0d923/960-api-services/src/main/java/io/harness/kustomize/KustomizeCommands.java#L10)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Name should be KustomizeCommand
</blockquote></details>

Done.

**Review by abosii-harness — DISMISSED** [review]
> Reviewed 3 of 3 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 11 file(s) to review. 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 11 file(s) to review. 1 of 2 approvals obtained.

**Review by GoelPratyush99 — DISMISSED** [review]
> Reviewed 1 of 7 files at r2, 1 of 2 files at r3, 4 of 7 files at r4, 2 of 2 files at r6, 3 of 3 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: :shipit: complete! ✓

**Review by GoelPratyush99 — APPROVED** [review]
> Reviewed 1 of 2 files at r10, 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r10, 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44419)* status: :shipit: complete! ✓

**Pull request merged**